src/Sulu/Bundle/MediaBundle/Resources/public/js/components/area-selection/main.js

```diff
@@ -0,0 +1,489 @@
+        instanceName: '',
+        image: null,
+        guideWidth: null,
+        guideHeight: null,
+        data: null
+    },
+
+    translations = {
+        minimumSizeReached: 'sulu-media.minimum-size-reached'
+    },
+
+    /**
+     * Raised when the component has been successfully initialized.
+     * @event sulu.area-selection.initialized
+     */
+    INITIALIZED = function() {
+        return createEventName.call(this, 'initialized');
+    },
+
+    /**
+     * Raised when the area the component selects has changed.
+     * @event sulu.area-selection.area-changed
+     */
+    AREA_CHANGED = function() {
+        return createEventName.call(this, 'area-changed');
+    },
+
+    /**
+     * Listens on and sets the guide dimensions of the component
+     *
+     * @param {Number} guideWidth The new guide-width
+     * @param {Number} guideHeight The new guide-height
+     * @param {Object} data The new data to use
+     *
+     * @event sulu.area-selection.change-guide-dimensions
+     */
+    SET_GUIDE_DIMENSIONS = function() {
+        return createEventName.call(this, 'set-guide-dimensions');
+    },
+
+    /**
+     * returns normalized event names
+     */
+    createEventName = function(postFix) {
+        return this.options.eventNamespace +
+            '.' + (!!this.options.instanceName ? this.options.instanceName + '.' : '') + postFix;
+    };
+
+    return {
+
+        initialize: function() {
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+
+            this.$frame = null;
+            this.$backdrop = null;
+
+            // The original width and height of the image.
+            this.originalWidth = null;
+            this.originalHeight = null;
+            this.guideWidth = this.options.guideWidth;
+            this.guideHeight = this.options.guideHeight;
+            this.physicalGuideWidth = null;
+            this.physicalGuideHeight = null;
+
+            // The actual area element with it's physical coordinates.
+            this.area = {
+                $el: null,
+                coordinates: {
+                    x: null,
+                    y: null,
+                    width: null,
+                    height: null
+                }
+            };
+
+            // The result of the component which gets computed from the physical coordinates of the
+            // area and the original dimensions of the image.
+            this.$el.data('area', {
+                x: null,
+                y: null,
+                width: null,
+                height: null
+            });
+
+            // Variables which are needed for dragging the area over the area
+            this.dragging = {
+                enabled: false,
+                clickOffsetLeft: 0,
+                clickOffsetTop: 0
+            };
+
+            // Variables which are needed for changing the width and height of the area
+            this.resizing = {
+                southEast: {
```

Comment:
We agreed on making the area only resizable in the bottom-right corner, as it drastically simplifies the component. I decided to make a sub-object for that to be more open for modification. If one want's to add resizing to other corners, code just needs to be added. Otherwise existing code would have to be changed.